### DISCUSSION TITLE
CloudSim-Plus build.xml

### DISCUSSION THREAD

**Discussion by baruach** [manoelcampos/cloudsim-plus#426] [Q&A] [answered]
I was told I needed to compile cloudsim-plus using ant after installing. Ant needs the build.xml file but it is not included in the downloaded zip. Where can I get the build.xml file? Search all over but couldn't find. Is it really needed?

Jopa

**Comment by manoelcampos** [comment, marked as answer]
CloudSim Plus is a maven project. You don’t need ant. Just open the project on any IDE an click the build button.

You can also open the project root dir at a terminal and type:

# Linux/macos
./mvnw clean install

# Windows
mvnw.cmd clean install

**Reply by baruach** [reply]
I've completed the Maven build with CloudSim Plus, installed all the dependencies, installed apache tomcat, setup the build path, etc. When I run a maven build i'm getting these warnings.  After many hours unable to resolved the issue. I will send future inquiries to the CloudSim Plus group site. Here is a summary of the warnings: 

[[1;34mINFO[m] Scanning for projects...
[[1;34mINFO[m] 
[[1;34mINFO[m] [1m-------------------< [0;36morg.cloudsimplus:cloudsimplus[0;1m >--------------------[m
[[1;34mINFO[m] [1mBuilding CloudSim Plus API 8.0.0[m
[[1;34mINFO[m] [1m--------------------------------[ jar ]---------------------------------[m
[[1;34mINFO[m] 
[[1;34mINFO[m] [1m--- [0;32mjacoco-maven-plugin:0.8.8:prepare-agent[m [1m(default)[m @ [36mcloudsimplus[0;1m ---[m
[[1;34mINFO[m] argLine set to "-javaagent:C:\\Users\\info\\.m2\\repository\\org\\jacoco\\org.jacoco.agent\\0.8.8\\org.jacoco.agent-0.8.8-runtime.jar=destfile=G:\\My Drive\\ITCOBRA\\0_TestWorkspace\\0_Projects\\8.0_Projects\\8Simulator-1\\cloudsimplus-8.0.0\\cloudsimplus-8.0.0\\target\\jacoco.exec"
[[1;34mINFO[m] 
[[1;34mINFO[m] [1m--- [0;32mlicense-maven-plugin:3.0:check[m [1m(default)[m @ [36mcloudsimplus[0;1m ---[m
[[1;34mINFO[m] Checking licenses...
[[1;33mWARNING[m] Missing header in: G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/allocationpolicies/migration/VmAllocationPolicyMigrationAbstract.java
[[1;33mWARNING[m] Missing header in: G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/allocationpolicies/VmAllocationPolicySimple.java
[[1;33mWARNING[m] Missing header in: G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-
[[1;34mINFO[m] 
[[1;34mINFO[m] [1m--- [0;32mmaven-resources-plugin:2.6:resources[m [1m(default-resources)[m @ [36mcloudsimplus[0;1m ---[m
[[1;34mINFO[m] Using 'UTF-8' encoding to copy filtered resources.
[[1;34mINFO[m] Copying 84 resources
[[1;34mINFO[m] 
[[1;34mINFO[m] [1m--- [0;32mmaven-compiler-plugin:3.8.1:compile[m [1m(default-compile)[m @ [36mcloudsimplus[0;1m ---[m
[[1;34mINFO[m] Changes detected - recompiling the module!
[[1;34mINFO[m] Compiling 341 source files to G:\My Drive\ITCOBRA\0_TestWorkspace\0_Projects\8.0_Projects\8Simulator-1\cloudsimplus-8.0.0\cloudsimplus-8.0.0\target\classes
[[1;34mINFO[m] /G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/distributions/ContinuousDistributionNull.java: G:\My Drive\ITCOBRA\0_TestWorkspace\0_Projects\8.0_Projects\8Simulator-1\cloudsimplus-8.0.0\cloudsimplus-8.0.0\src\main\java\org\cloudsimplus\distributions\ContinuousDistributionNull.java uses or overrides a deprecated API.
[[1;34mINFO[m] /G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/distributions/ContinuousDistributionNull.java: Recompile with -Xlint:deprecation for details.
[[1;34mINFO[m] /G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/resources/Resource.java: Some input files use unchecked or unsafe operations.
[[1;34mINFO[m] /G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/resources/Resource.java: Recompile with -Xlint:unchecked for details.
[[1;34mINFO[m] -------------------------------------------------------------
[[1;31mERROR[m] COMPILATION ERROR : 
[[1;34mINFO[m] -------------------------------------------------------------
[[1;31mERROR[m] /G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/core/CloudInformationService.java:[97,16] getId() in org.cloudsimplus.core.CloudInformationService cannot implement getId() in org.cloudsimplus.core.Identifiable
  return type int is not compatible with long
[[1;31mERROR[m] /G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/core/CloudInformationService.java:[96,5] method does not override or implement a method from a supertype
[[1;31mERROR[m] /G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/core/CloudInformationService.java:[98,20] cannot find symbol
  symbol: method getInstanceId()
[[1;34mINFO[m] 3 errors 
[[1;34mINFO[m] -------------------------------------------------------------
[[1;34mINFO[m] [1m------------------------------------------------------------------------[m
[[1;34mINFO[m] [1;31mBUILD FAILURE[m
[[1;34mINFO[m] [1m------------------------------------------------------------------------[m
[[1;34mINFO[m] Total time:  26.146 s
[[1;34mINFO[m] Finished at: 2023-03-29T11:10:53-07:00
[[1;34mINFO[m] [1m------------------------------------------------------------------------[m
[[1;31mERROR[m] Failed to execute goal [32morg.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile[m [1m(default-compile)[m on project [36mcloudsimplus[m: [1;31mCompilation failure[m: Compilation failure: 
[[1;31mERROR[m] /G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/core/CloudInformationService.java:[97,16] getId() in org.cloudsimplus.core.CloudInformationService cannot implement getId() in org.cloudsimplus.core.Identifiable
[[1;31mERROR[m]   return type int is not compatible with long
[[1;31mERROR[m] /G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/core/CloudInformationService.java:[96,5] method does not override or implement a method from a supertype
[[1;31mERROR[m] /G:/My Drive/ITCOBRA/0_TestWorkspace/0_Projects/8.0_Projects/8Simulator-1/cloudsimplus-8.0.0/cloudsimplus-8.0.0/src/main/java/org/cloudsimplus/core/CloudInformationService.java:[98,20] cannot find symbol
[[1;31mERROR[m]   symbol: method getInstanceId()
[[1;31mERROR[m] -> [1m[Help 1][m
[[1;31mERROR[m] 
[[1;31mERROR[m] To see the full stack trace of the errors, re-run Maven with the [1m-e[m switch.
[[1;31mERROR[m] Re-run Maven using the [1m-X[m switch to enable full debug logging.
[[1;31mERROR[m] 
[[1;31mERROR[m] For more information about the errors and possible solutions, please read the following articles:
[[1;31mERROR[m] [1m[Help 1][m http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

**Comment by baruach** [comment]
Thank you very much!

**Reply by ArkS0001** [reply]
how was error fixed?

**Comment by ethicalByte1443** [comment]
**NEED ATTENTION:**  can you share how was this error got fixed??

**Comment by ethicalByte1443** [comment]
![Screenshot 2024-05-05 143140](https://github.com/cloudsimplus/cloudsimplus/assets/135228041/7775da78-88e5-4c44-89de-2a400e49db0f)

**Reply by manoelcampos** [reply]
You’re probably using an incorrect JDK version

**Reply by ethicalByte1443** [reply]
i have used jdk 21 and also 17 but got no result
Can you please tell me which version you used to bring things out

**Reply by manoelcampos** [reply]
The required JDK version is 17, as stated in the readme. If you are compiling the code on the terminal, you must ensure you terminal is seeing that version. Execute javac -version to confirm that the current JDK version is 17.

**Reply by ethicalByte1443** [reply]
yeahh thanks it worked!!!!
